local_server/test/unit/common/test_api.py

```diff
@@ -677,8 +691,86 @@ def test_case(test, expected_code, original_data):
+        url = f"{self.URL_BASE}{endpoint}"
+        header = {"Accept": "application/octet-stream"}
+        result = self.session.get(url, headers=header)
+        self.assertEqual(result.status_code, HTTPStatus.OK)
+        self.assertEqual(result.headers["Content-Type"], "application/octet-stream")
+        df = decode_fbs.decode_matrix_FBS(result.content)
+        self.assertEqual(df["n_rows"], 2638)
+        self.assertEqual(df["n_cols"], 1)
+        self.assertEqual(df["col_idx"], ["summary test"])
+        self.assertAlmostEqual(df["columns"][0][0], -0.19863907)
+
+    def test_get_geneset_summary_check_tid(self):
+        # get the TID
+        result = self.session.get(f"{self.URL_BASE}genesets", headers={"Accept": "application/json"})
+        self.assertEqual(result.status_code, HTTPStatus.OK)
+        tid = result.json()["tid"]
+
+        # current tid
+        endpoint = f"geneset_summary?geneset_name=summary%20test&tid={tid}"
+        result = self.session.get(f"{self.URL_BASE}{endpoint}", headers={"Accept": "application/octet-stream"})
+        self.assertEqual(result.status_code, HTTPStatus.OK)
+
+        # future tid
+        endpoint = f"geneset_summary?geneset_name=summary%20test&tid={tid+1}"
+        result = self.session.get(f"{self.URL_BASE}{endpoint}", headers={"Accept": "application/octet-stream"})
+        self.assertEqual(result.status_code, HTTPStatus.NOT_FOUND)
+
+        # past tid
+        endpoint = f"geneset_summary?geneset_name=summary%20test&tid={tid-1}"
+        result = self.session.get(f"{self.URL_BASE}{endpoint}", headers={"Accept": "application/octet-stream"})
+        self.assertEqual(result.status_code, HTTPStatus.NOT_FOUND)
+
+        # No tid - ie, skip check
+        endpoint = "geneset_summary?geneset_name=summary%20test"
+        result = self.session.get(f"{self.URL_BASE}{endpoint}", headers={"Accept": "application/octet-stream"})
+        self.assertEqual(result.status_code, HTTPStatus.OK)
+
+    def test_get_geneset_summary_edge_cases(self):
+        # attempt to summarize _all_ genesets, including edge cases with zero or one gene
+        result = self.session.get(f"{self.URL_BASE}genesets", headers={"Accept": "application/json"})
+        self.assertEqual(result.status_code, HTTPStatus.OK)
+        geneset_names = [gs["geneset_name"] for gs in result.json()["genesets"]]
+
+        for gs in geneset_names:
+            endpoint = f"geneset_summary?geneset_name={gs}"
+            result = self.session.get(f"{self.URL_BASE}{endpoint}", headers={"Accept": "application/octet-stream"})
+            self.assertEqual(result.status_code, HTTPStatus.OK)
+            self.assertEqual(result.headers["Content-Type"], "application/octet-stream")
+            df = decode_fbs.decode_matrix_FBS(result.content)
+            self.assertEqual(df["n_rows"], 2638)
+            self.assertEqual(df["n_cols"], 1)
+            self.assertEqual(df["col_idx"], [gs])
+
+    def test_get_geneset_error_handling(self):
```

Comment:
I think missing a test case for `UnsupportedSummaryMethod`

Reply:
It is the last test in the list, where it asks for `method=NO_SUCH_METHOD`

Reply:
🤦 I see it now. Thanks. My squirrel brain just skimmed right over it.